src/DIRAC/Core/Tornado/Server/private/BaseRequestHandler.py

```diff
@@ -662,26 +670,46 @@ def on_finish(self):
         Called after the end of HTTP request.
         Log the request duration
         """
-        elapsedTime = 1000.0 * self.request.request_time()
+        elapsedTime = self.request.request_time()
+
         credentials = self.srv_getFormattedRemoteCredentials()
 
         argsString = f"OK {self._status_code}"
+        monitoringRetStatus = "Unknown"
         # Finish with DIRAC result
         if isReturnStructure(self.__result):
             if self.__result["OK"]:
                 argsString = "OK"
+                monitoringRetStatus = "OK"
             else:
                 argsString = f"ERROR: {self.__result['Message']}"
+                monitoringRetStatus = "ERROR"
                 if callStack := self.__result.pop("CallStack", None):
                     argsString += "\n" + "".join(callStack)
         # If bad HTTP status code
         if self._status_code >= 400:
             argsString = f"ERROR {self._status_code}: {self._reason}"
 
         self.log.notice(
-            "Returning response", f"{credentials} {self._fullComponentName} ({elapsedTime:.2f} ms) {argsString}"
+            "Returning response",
+            f"{credentials} {self._fullComponentName} ({1000.0 * elapsedTime:.2f} ms) {argsString}",
         )
 
+        if self.activityMonitoringReporter:
+            record = {
+                "timestamp": int(TimeUtilities.toEpochMilliSeconds()),
+                "Host": Network.getFQDN(),
+                "ServiceName": "_".join(self._fullComponentName.split("/")),
+                "Location": self.request.uri,
+                "ResponseTime": elapsedTime,
+                # Take the method name from the POST call
+                "MethodName": self.request.arguments.get("method", ["Unknown"])[0],
```

Comment:
Should we include the HTTP verb (POST/GET/...) as well? Either in the method name or as a separate field?

Reply:
That was my original idea, but it will only ever be different from a `POST` when it's using `TornadoREST`. And there, it gets reaaaaaally messy..... so for the moment, that works with our standard `TornadoServices`, not for the `TornadoRest`

Reply:
Ah yeah, the more I think about it the messier it gets...